client/lib/signup/step-actions/index.js

```diff
@@ -227,6 +228,15 @@ function getNewSiteParams( {
 		newSiteParams.options.in_page_builder = true;
 	}
 
+	if ( selectedDesign ) {
+		// If there's a selected design, it means we're in "with_design_picker" flow.
```

Comment:
This comment could easily become outdated if we introduce other flows with the `design` step — what if we changed it to something like

```suggestion
		// If there's a selected design, it means that the current flow contains the "design" step.
```